samples/switchless/README.md

```diff
@@ -0,0 +1,184 @@
+to the worker threads is actually 0, and the minimum of 1) and 2) would be 0. In this case, we recommend setting
+the number of host worker threads to 1 nevertheless, to ensure switchless calls are serviced by at least one thread.
+
+The above recommendation may change when we modify the behavior of worker threads in the future.
+
+## About the EDL
+
+In this sample, we pretend the enclave doesn't know addition. It relies on a host function `host_increment` to
+increment a number by 1, and repeat calling it `N` times to add `N` to a given number. Since `host_increment` is
+short and called frequently, it is appropriate to make it a switchless function.
+
+First we need to define the functions we want to call between the host and the enclave. To do this we create a `switchless.edl` file:
+
+```edl
+enclave {
+    trusted {
+        public void enclave_add_N([in, out] int* m, int n);
+    };
+
+    untrusted {
+        void host_increment([in, out] int* m) transition_using_threads;
+    };
+};
+```
+
+Function `host_increment`'s declaration ends with keyword `transition_using_threads`, indicating it should be called switchlessly at run time. However, this a best-effort directive. OpenEnclave runtime may still choose to fall back to a tradition OCALL if switchless call resources are unavailable, e.g., the enclave is not configured as switchless-capable, or the host worker threads are busy servicing other switchless OCALLs.
+
+To generate the functions with the marshaling code, the `oeedger8r` tool is called in both the host and enclave directories from their Makefiles. For example:
+
+```bash
+cd host
+oeedger8r ../switchless.edl --untrusted --experimental
+```
+
+`oeedger8r` needs the command line flag `--experimental` to be able to recognize the keyword `transition_using_threads`.
+
+## About the host
+
+The host first defines a structure specifically for configuring switchless calls. In this case, we specify the
+first field `1` as the number of host worker threads for switchless OCALLs. In this example, 1) There is at most
+1 enclave thread all the time, and 2) The number of cores available to the host worker threads is  1 (assuming a
+2-core machine) or 3 (assuming a 4-core machine). In any case, The minimum of 1) and 2) is 1, which we choose to
+be the number of host worker threads. The 2nd field specifies the number of enclave threads for switchless ECALLs.
+Since switchless ECALL is not yet implemented, we require the 2nd field to be `0`.
+
+```c
+oe_enclave_config_context_switchless_t config = {1, 0};
+```
+
+The host then puts the structure address and the configuration type in an array of configurations for the enclave to be created. Even though we only have one configuration (for switchless) for the enclave, we'd like the flexibility of adding more than one configuration (with different types) for an enclave in the future.
```

Comment:
I don't think "more than one configuration" is what we mean, since in normal English, configurations are alternatives, meaning you only use one configuration at a time (e.g., configurations of a project in Visual Studio, etc.).   Instead I think they are configuration options, or configuration elements, or configuration parameters.

This may mean the variable name and structure name are both probably needing to change for clarity.

Reply:
@mikbras once mentioned using `settings` instead of `configuration`. Inherently `settings` implies plurality. Does that sound a better choice? But I am not sure whether we call each structure in the array a `setting` or `settings`. It's for a single feature, but could have multiple knobs.

Reply:
oe_enclave_setting_context_switchless_t setting = {1, 0};
would be fine with me.

Reply:
Configuration Parameters looks good to me. So "we'd like the flexibility of adding more than one configuration parameter".

Reply:
Opened issue #2202 to track @dthaler 's recommendation for the change in variable name